### PR TITLE
Fix linting errors

### PR DESCRIPTION
**Pull request by argahsuknesib**
The pull request corresponds to the issue [30](https://github.com/SolidLabResearch/solid-stream-aggregator/issues/30) on the repository.

I fixed the linting issues in the repository. The unit tests are still a work in progress which I will fix and add with another pull request after the linting errors are reviewed.

Running the command,
```bash
npm run lint:ts
```
currently gives, 
```bash
✖ 267 problems (0 errors, 267 warnings)
```
The warnings are documentation warnings with JSDoc which aren't important to resolve at the current moment (please correct me on this) and will be resolved in future.

### CODE REVIEW

**Review by pheyvaer — CHANGES_REQUESTED** [review]
The folder `.idea` is personal to your setup and doesn't belong in the repo.

**Comment by argahsuknesib** [comment]
Thanks for pointing it out @pheyvaer, I changed the IDE but forgot to add the personal setup of the IDE into the .gitignore file.

**Comment by pheyvaer** [comment]
> The warnings are documentation warnings with JSDoc which aren't important to resolve at the current moment (please correct me on this) and will be resolved in future.

The documentation is the main reason why I added linting 😄  So yes they have to be fixed.

**Comment by argahsuknesib** [comment]
Running,
```bash
npm run lint:ts
```
will give no errors or warnings now.

**Comment by pheyvaer** [comment]
@argahsuknesib We also need to docs for the classes and their methods. Can you fix that as well? I updated the lint config for that.

**Comment by argahsuknesib** [comment]
I got another 126 JSDoc comment warnings, I will get rid of the warnings with another commit and reassign you to review. Thanks @pheyvaer !

**Review by pheyvaer — CHANGES_REQUESTED** [review]

**Comment by pheyvaer** [comment]
> I got another 126 JSDoc comment warnings, I will get rid of the warnings with another commit and reassign you to review. Thanks @pheyvaer !

Ok, great! I also added some minor comments of things that I noticed.

File: logs/aggregation.log
[the remark this answers is not in the document]
Comment (argahsuknesib) [reply logs/aggregation.log:1]: I will add them to .gitignore

File: scratch/test.ts
[the remark this answers is not in the document]
Comment (argahsuknesib) [reply scratch/test.ts:1]: The scratch files are created by me, in which I test some code. But they don't belong in the repo and I added them to .gitignore.

File: src/service/query-registry/QueryRegistry.test.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+    it('initializing the QueryRegistry', () => {
+        expect(query_registry).toBeInstanceOf(QueryRegistry);
+    });
+    it(`adding a query to the registry`, async () => {
+        expect(await query_registry.add_query_in_registry(rspql_query, logger)).toBe(true);
+        query_registry.delete_all_queries_from_the_registry();
+    });
+
+    it('delete_all_queries_from_the_registry', async () => {
```
[the remark this answers is not in the document]
Comment (argahsuknesib) [reply src/service/query-registry/QueryRegistry.test.ts:33]: ok thanks, will do.

File: src/server/HTTPServer.ts [hunk 2]
```diff
@@ -40,19 +39,16 @@ export class HTTPServer {
...
+        let body: string = '';
         switch (req.method) {
             case "GET":
-                const latest_minutes = parsed_url.query.latest_minutes;
-                GETHandler.handle(req, res, this.solid_server_url, this.query_registry, this.endpoint_queries, latest_minutes, this.logger);
+                GETHandler.handle(req, res);
                 res.end();
                 break;
             case "POST":
                 // TODO : bug that the notification is sent more than once from the solid server.
```
[the remark this answers is not in the document]
Comment (argahsuknesib) [reply src/server/HTTPServer.ts:65]: It is not, I will add it into the issues.

**Comment by argahsuknesib** [comment]
I am working on the tests right now @pheyvaer, should I re-request a review when the coverage is 100%?

**Comment by pheyvaer** [comment]
> I am working on the tests right now @pheyvaer, should I re-request a review when the coverage is 100%? (as the linting errors are fixed now)

This PR is only for linting errors. So if you are changing/adding tests then this should happen in a separate branch. The linting errors for the tests that are already in this PR should be covered yes.

File: src/utils/ldes-in-ldp/EventSource.ts [hunk 3]
```diff
@@ -314,18 +299,38 @@ export async function* readPageRateLimited(ldes: LDESinLDP, fragment_url: string
...
- * @param resource_identifier
- * @param rate_limit_comm
+ * @param {LDESinLDP} ldes - The LDES in LDP object.
+ * @param {string} resource_identifier - The identifier of the resource to be read.
+ * @param {RateLimitedLDPCommunication} rate_limit_comm - The rate limited LDP communication object to communicate to the LDP.
+ * @returns {Promise<Store>} - Returns the resource as an N3 Store.
  */
 export async function readRateLimited(ldes: LDESinLDP, resource_identifier: string, rate_limit_comm: RateLimitedLDPCommunication) {
+    try {
+        // TODO : check for headers, as well as error handling. check if you can increase the timeout for the get request as some resources might take longer to load (due to large files, slow server, etc.)
```
Comment (pheyvaer) [inline src/utils/ldes-in-ldp/EventSource.ts:313]: Can you add issue for this as well?

Comment (argahsuknesib) [reply src/utils/ldes-in-ldp/EventSource.ts:313]: done for this as well.

File: src/service/aggregator/AggregatorInstantiator.ts [hunk 4]
```diff
@@ -78,6 +103,16 @@ export class AggregatorInstantiator {
     }
 
     // TODO : add extra projection variables to the aggregation event.
```
[the remark this answers is not in the document]
Comment (argahsuknesib) [reply src/service/aggregator/AggregatorInstantiator.ts:105]: done

**Review by pheyvaer — APPROVED** [review]

**Pull request merged**